modules/extension/transform/src/main/java/org/geotools/data/transform/TransformFeatureCollection.java

```diff
@@ -206,4 +208,25 @@ public SimpleFeatureCollection subCollection(Filter filter) {
         }
         return new TransformFeatureCollection(source, transformer, q);
     }
+
+    @Override
+    public void accepts(
+            org.opengis.feature.FeatureVisitor visitor, org.opengis.util.ProgressListener progress)
+            throws IOException {
+        Name typeName = transformer.getSource().getName();
+
+        if (typeName != null) {
+            try {
+                SimpleFeatureSource fs =
+                        (SimpleFeatureSource) source.getDataStore().getFeatureSource(typeName);
+                if (fs instanceof ContentFeatureSource) {
```

Comment:
No actual need to check if it's a content feature source, SimpleFeatureSource has the method you need too